File: golang/src/domain/dimensao.go
package domain

type Dimensao struct {
	Altura       float64
	Largura      float64
	Profundidade float64
}

func (dimensao Dimensao) GetVolume() float64 {
	alturaEmMetros := converteMetros(dimensao.Altura)
	larguraEmMetros := converteMetros(dimensao.Largura)
	profundidadeEmMetros := converteMetros(dimensao.Profundidade)

	return alturaEmMetros * larguraEmMetros * profundidadeEmMetros
}

func converteMetros(distancia float64) float64 {
	return distancia / 100
}

File: node/src/domain/Dimensao.ts

export default class Dimensao{

    constructor(readonly altura: number, readonly largura: number, readonly profundidade: number){}

    public getVolume(){
        const alturaEmMetros = this.converterMetros(this.altura)
        const larguraEmMetros = this.converterMetros(this.largura)
        const profundidadeEmMetros = this.converterMetros(this.profundidade)
        return (alturaEmMetros * larguraEmMetros * profundidadeEmMetros)
    }

    private converterMetros(valor: number): number{
        return valor/100
    }
}